src/App.js

```diff
@@ -1,13 +1,21 @@
-import React from 'react';
-import { Route, Switch } from 'react-router-dom';
+import React, { useState } from 'react';
+import { Route, Switch, Redirect } from 'react-router-dom';
 import './App.css';
-import Test from './components/Test';
 import Navigation from './components/Navigation';
 import AddItem from './components/AddItem';
 import List from './components/List';
 import Welcome from './components/Welcome';
+import getToken from './lib/tokens';
 
 function App() {
+  const [token, setToken] = useState(localStorage.getItem('userToken') || '');
+
+  const handleClick = () => {
+    const token = getToken();
+    localStorage.setItem('userToken', token);
+    setToken(token);
+  };
+
   return (
     <div className="App">
       <Switch>
```

Comment:
```suggestion
      <Switch>
        <Route path="/list">
          {token ? <List /> : <Welcome onClick={handleClick} />}
        </Route>

        <Route path="/add-item">
          {token ? <AddItem /> : <Welcome onClick={handleClick} />}
        </Route>

        <Route path="/">
          {token ? <Redirect to="/list" /> : <Welcome onClick={handleClick} />}
        </Route>
      </Switch>
```

Reply:
I think we should render the Welcome component if the user does not already have a user token. I think there are probably ways we could clean this up so we don't duplicate code.

Reply:
@rachelrohrbach I like your suggestion! Instead of the welcome component in those top two ternaries (lines 23 and 27), I think a `<Redirect to="/" />` would do the trick. I imagine it could be even fancier, but I _think_ that would get the job done, so to speak. Otherwise, without a token, the user could be at the `/list` in the url, but it would have the `Welcome` screen.

Reply:
Awesome! Thank you Rachel. This is a great suggestion. I'll get together with Viet on this! Great suggestion and we can check in with our mentors regarding ways to reduce duplicate code!

Reply:
Thanks @rachelrohrbach and @laurenmbeatty for the feedback! Thinking about how we implement this 🤔

Reply:
I feel like I've learned so much working on this issue and reading through everyone's feedback. Thanks team!